File: zul/src/com/potix/zul/html/Toolbar.java
/* Toolbar.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Thu Jun 23 11:33:31     2005, Created by tomyeh@potix.com
}}IS_NOTE

Copyright (C) 2005 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 2.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
*/
package com.potix.zul.html;

import com.potix.lang.Objects;

import com.potix.zk.ui.WrongValueException;
import com.potix.zul.html.impl.XulElement;

/**
 * A toolbar.
 *
 * <p>Default {@link #getSclass}: toolbox.
 *
 * @author <a href="mailto:tomyeh@potix.com">tomyeh@potix.com</a>
 */
public class Toolbar extends XulElement {
	private String _orient = "horizontal";

	public Toolbar() {
		setSclass("toolbar");
	}

	/** Returns the orient.
	 * <p>Default: "horizontal".
	 */
	public String getOrient() {
		return _orient;
	}
	/** Sets the orient.
	 * @param orient either "horizontal" or "vertical".
	 */
	public void setOrient(String orient) throws WrongValueException {
		if (!"horizontal".equals(orient) && !"vertical".equals(orient))
			throw new WrongValueException("orient cannot be "+orient);

		if (!Objects.equals(_orient, orient)) {
			_orient = orient;
			invalidate(INNER);
		}
	}
}

File: zul/src/archive/web/js/zul/wgt/mold/toolbar.js
/* toolbar.js

	Purpose:

	Description:

	History:
		Sat Dec 24 12:58:43     2008, Created by Flyworld

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
function (out) {
	var uuid = this.uuid;

	out.push('<div ', this.domAttrs_(), '><div id="', uuid, '-cave"',
			' class="', this.$s('content'), ' ', this.$s(this.getAlign()), '" role="none" >');

	this._tabindexTargetChild = null;
	for (var w = this.firstChild; w; w = w.nextSibling) {
		w.redraw(out);
		// ZK-1706: the width of empty space does not always equal 3px in ie9
		if ('horizontal' == this.getOrient() && zk.ie9)
			out.push('<span></span>');
	}
	out.push('</div>');

	if (this.isOverflowPopup()) {
		out.push('<i id="', uuid, '-overflowpopup-button"',
				' class="', this.$s('overflowpopup-button'), ' z-icon-ellipsis-h z-icon-fw" aria-hidden="true"></i>');
		out.push('<div id="', uuid, '-pp" class="', this.$s('popup'), ' ', this.$s('popup-close'), '"></div>');
	}
	out.push('<div class="z-clear" aria-hidden="true"></div></div>');
}
